### ISSUE TITLE
Research improvement of Error handling and output

### ISSUE THREAD

**Issue by ostowe** [alleyinteractive/huron#60]
A lot of improvement can be made in the way we're handling and notifying the user of errors. My initial thoughts on this are that we should have an error class for both front-end and CLI, and we should improve and increase the number of error messages.

**Comment by ellm** [comment]
@ostowe If possible, I'd be interested in taking this on. I'm still trying to familiarize myself with the codebase and I think this would be a great way to dig deeper.

**Comment by ostowe** [comment]
@ellm yeah, that'd be great! I say go for it.

**Comment by ostowe** [comment]
That'd be great, I say go for it!

**Comment by ellm** [comment]
Thanks @ostowe - I've been looking through the code and noticed that some methods are missing `@return` documentation. I think while I am looking into the error handling, I'm gonna try to get any missing return documentation in place.

**Comment by ostowe** [comment]
@ellm absolutely, go ahead and fix what you find, just be sure to run JSDoc once you're done to verify it's incorporated into the docs as expected. Thanks!
